scrapinghub/client/utils.py

```diff
@@ -136,40 +137,31 @@ class _MappingProxy(_ProxyBase):
 
     def __init__(self, cls, client, key):
         super(_MappingProxy, self).__init__(cls, client, key)
-        methods = ['ignore_fields', 'expire', 'save', 'liveget']
-        self._proxy_methods(methods)
-        self.iterkeys = self.__iter__
-
-    def __str__(self):
-        return str(self._origin._data)
-
-    def __repr__(self):
-        return '{}({})'.format(self.__class__.__name__,
-                               repr(self._origin._data))
-
-    def __getitem__(self, key):
-        return self._origin[key]
-
-    def __setitem__(self, key, value):
-        self._origin[key] = value
 
-    def __delitem__(self, key):
-        del self._origin[key]
+        # cls_methods methods will be imported from origin class
+        origin_cls_methods = ['__iter__', '__len__', '__str__', '__repr__',
+                              '__getitem__', '__setitem__', '__delitem__']
 
-    def __iter__(self):
-        return iter(self._origin)
+        other_attrs = ['ignore_fields', 'expire', '_data', '_deleted',
+                       'save', 'liveget', ('iter', '__iter__')]
 
-    def __len__(self):
-        return len(self._origin)
+        # blacklisted attributes should be just skipped on proxying
+        blacklisted_attrs = ['__class__', '__init__', '__str__', '__repr__']
 
-    def keys(self):
-        return list(self.iterkeys())
+        # proxy all the main methods from MutableMapping
+        methods = [method for method in dir(MutableMapping)
+                   if callable(getattr(MutableMapping, method))
+                   and method not in origin_cls_methods
+                   and method not in blacklisted_attrs]
+        proxy_methods(MutableMapping, self.__class__, methods, force=True)
```

Comment:
IMO you don't need to separate these methods from MutableMapping methods - as `self._origin` is an instance of a subclass of a MutableMapping you'll need only build a list of methods to proxy as 

```
mutable_mapping_methods = {method for method in dir(MutableMapping) if callable(getattr(MutableMapping, method))}
blacklisted_methods = {'__class__', '__init__', '__str__', '__repr__'}
resource_type_methods = {'ignore_fields', 'expire', '_data', '_deleted', 'save', 'liveget', ('iter', '__iter__')}
methods_to_proxy = (mutable_mapping_methods | resource_type_methods) - blacklisted_methods
```

and then proxy them **all** to the origin

```
proxy_methods(self._origin, self, methods_to_proxy)
```